library/src/camera_impl.cxx

```diff
@@ -0,0 +1,241 @@
+}
+
+//----------------------------------------------------------------------------
+camera::vector_t camera_impl::getFocalPoint()
+{
+  vector_t foc;
+  vtkCamera* cam = this->GetVTKCamera();
+  cam->GetFocalPoint(foc.data());
+  return foc;
+}
+
+//----------------------------------------------------------------------------
+void camera_impl::setViewUp(const vector_t& up)
+{
+  vtkCamera* cam = this->GetVTKCamera();
+  cam->SetViewUp(up.data());
+  cam->OrthogonalizeViewUp();
+  this->Internals->VTKRenderer->ResetCameraClippingRange();
+}
+
+//----------------------------------------------------------------------------
+camera::vector_t camera_impl::getViewUp()
+{
+  vector_t up;
+  vtkCamera* cam = this->GetVTKCamera();
+  cam->GetViewUp(up.data());
+  return up;
+}
+
+//----------------------------------------------------------------------------
+void camera_impl::setViewAngle(const double& angle)
+{
+  vtkCamera* cam = this->GetVTKCamera();
+  cam->SetViewAngle(angle);
+  cam->OrthogonalizeViewUp();
+  this->Internals->VTKRenderer->ResetCameraClippingRange();
+}
+
+//----------------------------------------------------------------------------
+double camera_impl::getViewAngle()
+{
+  vtkCamera* cam = this->GetVTKCamera();
+  return cam->GetViewAngle();
+}
+
+//----------------------------------------------------------------------------
+void camera_impl::setViewMatrix(const matrix_t& matrix)
+{
+  vtkCamera* cam = this->GetVTKCamera();
+  cam->SetPosition(matrix[12], matrix[13], matrix[14]);
+  cam->SetFocalPoint(matrix[8] + matrix[12], matrix[9] + matrix[13], matrix[10] + matrix[14]);
+  cam->SetViewUp(matrix[4], matrix[5], matrix[6]);
+//   cam->OrthogonalizeViewUp(); TODO NOT NEEDED ?
+  this->Internals->VTKRenderer->ResetCameraClippingRange();
+}
+
+//----------------------------------------------------------------------------
+camera::matrix_t camera_impl::getViewMatrix()
+{
+  vtkCamera* cam = this->GetVTKCamera();
+  matrix_t arr;
+#if 0
+  // Is this correct ?
+  double pos[3];
+  cam->GetPosition(pos);
+  double foc[3];
+  cam->GetFocalPoint(foc);
+  double up[3];
+  cam->GetViewUp(up);
+  double to[3];
+  vtkMath::Subtract(foc, pos, to);
+  double right[3];
+  vtkMath::Cross(to, up, right);
+
+  // TODO with normalize, the focal point is not at the right location
+  // with a get/set
+  vtkMath::Normalize(right);
+  std::move(std::begin(right), std::end(right), arr.begin());
+  arr[3] = 0;
+  vtkMath::Normalize(up);
+  std::move(std::begin(up), std::end(up), arr.begin() + 4);
+  arr[7] = 0;
+  vtkMath::Normalize(to);
+  std::move(std::begin(to), std::end(to), arr.begin() + 8);
+  arr[11] = 0;
+  std::move(std::begin(pos), std::end(pos), arr.begin() + 12);
+  arr[15] = 1;
+#else
+  // TODO  this implementation seems better
+  vtkMatrix4x4* mat = cam->GetModelViewTransformMatrix();
+  mat->Transpose();
+  double* data = mat->GetData();
+  std::move(data, data + 16, arr.begin());
```

Comment:
does that looks correct to you @Meakk ?

Reply:
Hmmm. Not sure why it's called `ModelViewTransform`, we do not care about the model transform. Anyway, yes it looks correct.  
However, make sure the call to `Transpose()` does not modify the matrix inside VTK.

Reply:
see the doc here:

https://vtk.org/doc/nightly/html/classvtkCamera.html#a40f9e41a86e5b7c94fe171e7b9193673

Ok I will check